user/service.py

```diff
@@ -18,26 +22,26 @@ def generate_salt() -> str:
     """Generate a random salt."""
     return str(uuid.uuid4())
 
-def create(username: str = None, password: str = None) -> int:
+def create(username: str, password: str) -> Optional[str]:
     """Create a new user."""
     if user_dao.get_by_username(username) is not None:
-        return -1
+        return None
 
     salt = generate_salt()
     return user_dao.create(username, hash_password(password + salt), salt)
 
-def update(user_id: int, username: str = None, password: str = None) -> bool:
+def update(user_id: str, username: str, password: str) -> bool:
```

Comment:
if we require these from the API consumer anyway I don't see the advantage of making these null by default. the code below (specifically the `password + salt` bit) seemed to assume these were non-null anyway so I just got rid of the default argument.

if however we want to make it possible to update some but not all fields in a single request I wonder if it might be cleaner to accept a JSON object of attributes/new values?